### PR TITLE
Feature/pm 152 connect wallet to show uport modal

### PR DESCRIPTION
**Pull request by mmvsha**
This PR moves all the provider integrations from WalletIntegrationProvider to a middleware, which allows us to re-init the integrations

### CODE REVIEW

File: src/integrations/injectedWeb3.js [hunk 1]
```diff
@@ -4,118 +4,127 @@ import { ETHEREUM_NETWORK, ETHEREUM_NETWORK_IDS } from 'integrations/constants'
...
 
 class InjectedWeb3 {
-    runProviderUpdate() {}
-    runProviderRegister() {}
+  runProviderUpdate() {}
+  runProviderRegister() {}
 
-    constructor(enableWatcher = true, defaultTimeout = 10000) {
-        this.defaultTimeout = defaultTimeout
+  constructor(enableWatcher = true, defaultTimeout = 10000) {
```
Comment (andre-meyer) [inline src/integrations/injectedWeb3.js]: These parameters really frustrate me. Can you make them static properties like the `priority` property?

File: src/integrations/injectedWeb3.js [hunk 2]
```diff
@@ -4,118 +4,127 @@ import { ETHEREUM_NETWORK, ETHEREUM_NETWORK_IDS } from 'integrations/constants'
 import { weiToEth, promisify } from 'utils/helpers'
 
 class InjectedWeb3 {
```
Comment (andre-meyer) [inline src/integrations/injectedWeb3.js]: I think now is also a good time to clean up this base class. I think this name is not fitting anymore. Most of the providers actually access web3 their own way. The name should suggest that this is the base class for all providers. `BaseIntegration`?

**Comment by mmvsha** [comment]
@andre-meyer I think integrations folder is a well placement and doesn't need to be changed. Agree with other refactors

**Comment by andre-meyer** [comment]
Ok! 👌

File: src/integrations/injectedWeb3.js (hunk 1 above)
Comment (apanizo) [reply src/integrations/injectedWeb3.js]: I kind of disagree on that. JS classes are an attempt to transform the dynamic objects stuff into a static one in order to do "easier" devs life, but it has many un-gotchas. Especially if you have some background on strongly typed class oriented languages like JAVA.
```
InjectedWeb3.myStaticProperty <--- works
const moe = new InjectedWeb3()
moe.myStaticProperty <---- does not work
```
If you have decided to create classes and delegate behaviour for modeling "real" case scenarios, if we are using some variables to tweak that behaviour they should be in the constructor.

If you think I have missed something let me know.

Comment (mmvsha) [reply src/integrations/injectedWeb3.js]: @apanizo but these properties (enableWatcher, defaultTimeout) are unigue for every instance. Why we cannot make it static?

Comment (andre-meyer) [reply src/integrations/injectedWeb3.js]: And we use the classes as singletons. `export default new uPort()` it's really difficult to see where they come from, and we have this line in all integrations: `super(false, 0)` which is very unreadable and confusing.

Also I think none of us have a background in Java 😄 I think it's logical that you don't access the static properties in the instances

Comment (andre-meyer) [reply src/integrations/injectedWeb3.js]: Ah but we'll need it in the BaseIntegration. So it'll be `this.constructor.watcherEnabled`, which is a bit ugly. Wdyt?

Comment (andre-meyer) [reply src/integrations/injectedWeb3.js]: Nevermind, we'll keep the static variables to the class. That way it's obvious how you need to reference them

File: src/integrations/baseIntegration.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+import { weiToEth, promisify } from 'utils/helpers'
+
+class BaseIntegration {
+  runProviderUpdate() {}
+  runProviderRegister() {}
+  enableWatcher = true
+  defaultTimeout = 1000
+
+  constructor(enableWatcher = this.enableWatcher, defaultTimeout = this.defaultTimeout) {
```
Comment (andre-meyer) [inline src/integrations/baseIntegration.js]: Can we use an object here? And add a jsdoc to document the parameters

**Pull request merged**